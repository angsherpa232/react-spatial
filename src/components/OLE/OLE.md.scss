.tm-ole-example {
  position: relative;

  .tm-ole {
    position: absolute;
    top: 10px;
    right: 10px;

    button {
      background: #fafafa;
      border: 0;
      padding: 5px;
      width: 45px;
      height: 45px;
      box-shadow: 0 3px 3px 0 rgba(0, 0, 0, 0.2);

      &.active {
        box-shadow: 0 4px 4px 0 rgba(0, 0, 0, 0.3);
        color: #5c5c5c;
        filter: brightness(90%);
      }
    }

    img {
      width: 35px;
      height: 35px;
    }
  }
}
