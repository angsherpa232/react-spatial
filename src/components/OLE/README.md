#

This demonstrates the use of OLE.

```jsx
import React from 'react';
import BasicMap from 'react-spatial/components/BasicMap';
import ConfigReader from 'react-spatial/ConfigReader';
import Popup from 'react-spatial/components/Popup';
import ResizeHandler from 'react-spatial/components/ResizeHandler';
import Layer from 'react-spatial/Layer';
import VectorLayer from 'react-spatial/VectorLayer';
import OLMap from 'ol/Map';
import Feature from 'ol/Feature';
import Point from 'ol/geom/Point';
import VectorSource from 'ol/source/Vector';
import TileLayer from 'ol/layer/Tile';
import TileGrid from 'ol/tilegrid/TileGrid';
import TileImageSource from 'ol/source/TileImage';
import { getCenter } from 'ol/extent';
import Style from 'ol/style/Style';
import Circle from 'ol/style/Circle';
import Fill from 'ol/style/Fill';
import OLE from 'react-spatial/components/OLE';
import OSM, {ATTRIBUTION} from 'ol/source/OSM.js';
import 'ol/ol.css';
import './OLE.md.scss';

class OLEExample extends React.Component {
  constructor(props) {
    super(props);

    this.map = new OLMap();

    this.layers = [
      new Layer({
        olLayer:new TileLayer({
          source: new OSM()
        })
      }),
      new VectorLayer({
        source: new VectorSource(),
      })
    ];
  }

  render() {
    return (
      <div className="tm-ole-example">
        <BasicMap
          map={this.map}
          layers={this.layers}
        />
        <OLE
          map={this.map}
          layer={this.layers[1]}
          cad
          drawPoint
          drawLineString
          drawPolygon
          move
          rotate
          modify
          del
          buffer
          union
          intersection
          difference
        />
      </div>
    );
  }
}

<OLEExample />;
```
