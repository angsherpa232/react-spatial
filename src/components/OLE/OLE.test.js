import 'jest-canvas-mock';
import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import VectorSource from 'ol/source/Vector';
import OLMap from 'ol/Map';
import View from 'ol/View';
import VectorLayer from '../../VectorLayer';
import OLE from '.';

configure({ adapter: new Adapter() });

const map = new OLMap({ view: new View() });
const vectorLayer = new VectorLayer({
  source: new VectorSource({
    features: [],
  }),
});

describe('OLE', () => {
  beforeEach(() => {
    map.setTarget(document.createElement('div'));
  });

  test('matches snapshots', () => {
    window.ole = 'fgjkfgio';
    const component = renderer.create(<OLE map={map} layer={vectorLayer} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('creates Editor with all controls with no error', () => {
    window.ole = 'fgjkfgio';
    mount(
      <OLE
        map={map}
        layer={vectorLayer}
        cad
        drawPoint
        drawCustoms={[{}]}
        drawLineString
        drawPolygon
        move
        rotate
        modify
        del
        buffer
        union
        intersection
        difference
      />,
    );
  });

  test('creates Editor without controls with no error', () => {
    window.ole = 'fgjkfgio';
    mount(
      <OLE
        map={map}
        layer={vectorLayer}
        cad={false}
        drawPoint={false}
        drawLineString={false}
        drawPolygon={false}
        move={false}
        rotate={false}
        modify={false}
        del={false}
        buffer={false}
        union={false}
        intersection={false}
        difference={false}
      />,
    );
  });
});
