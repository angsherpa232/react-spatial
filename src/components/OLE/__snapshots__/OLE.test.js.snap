// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OLE matches snapshots 1`] = `
<div
  className="tm-ole"
/>
`;
